### PR TITLE
search-modals

### PR DESCRIPTION
**Pull request by matthewwrobel**
-implemented question search
-implemented add a question and add an answer forms

@alizehrehman @tchitrakorn i'm looking forward to your feedback--thanks in advance!

### CODE REVIEW

File: client/App.jsx [hunk 1]
```diff
@@ -10,6 +10,7 @@ class App extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
+      name: 'Morning Joggers',
```
Comment (alizehrehman) [inline client/App.jsx:13]: We'll need to move the api calls to App later to ensure you can pass the name down as props

File: client/questions_answers/QuestionsAnswers.jsx [hunk 2]
```diff
@@ -25,24 +44,35 @@ class QuestionsAnswers extends React.Component {
...
         this.setState({
           showMoreAnsweredQuestionsButton: true,
+          showSearch: true,
           nextTwoQuestions: nextTwoQuestions,
           questions: firstTwoQuestions
         });
 
       });
-
-    this.updateQuestionsList = this.updateQuestionsList.bind(this);
```
Comment (alizehrehman) [inline client/questions_answers/QuestionsAnswers.jsx:41]: Why did you include this here vs in the constructor?

File: client/questions_answers/QuestionsAnswers.jsx [hunk 3]
```diff
@@ -61,14 +91,71 @@ class QuestionsAnswers extends React.Component {
...
+
   render() {
 
     return (
       <div className="questions-answers component">
         <div> {`QUESTIONS & ANSWERS`} </div>
-        <QuestionsList questions={this.state.questions}/>
-        {this.state.showMoreAnsweredQuestionsButton && <button onClick={this.updateQuestionsList}>MORE ANSWERED QUESTIONS</button>}
-        <button onClick={() => console.log('add a new question by clicking this button')}>ADD A QUESTION</button>
+        {this.state.showSearch && <Search query={this.state.query} handleSearch={this.handleSearch}/>}
```
Comment (alizehrehman) [inline client/questions_answers/QuestionsAnswers.jsx:154]: Very clever! Really like this!

**Review by alizehrehman — APPROVED** [review]
@matthewwrobel Looks great! Just made a few comments to consider. Approved! Merge away!

File: client/questions_answers/QuestionsAnswers.jsx (hunk 2 above)
Comment (matthewwrobel) [reply client/questions_answers/QuestionsAnswers.jsx:41]: Lets discuss tonight in standup...

Comment (alizehrehman) [reply client/questions_answers/QuestionsAnswers.jsx:41]: Sounds good!

**Pull request merged**